spec/models/condo_spec.rb

```diff
@@ -1,5 +1,65 @@
 require 'rails_helper'
 
 RSpec.describe Condo, type: :model do
-  # To do
+  describe '#valid' do
+    context 'presence' do
+      it 'false when params are empty' do
+        condo = build(:condo, name: nil, registration_number: nil)
+
+        expect(condo).not_to be_valid
+        expect(condo.errors).to include(:name)
+        expect(condo.errors).to include(:registration_number)
+      end
+    end
+
+    context 'uniqueness' do
+      it 'false when registration number is already in use' do
+        create(:condo, registration_number: '38.352.640/0001-33')
+        condo = build(:condo, registration_number: '38.352.640/0001-33')
+        condo.valid?
+        expect(condo.errors.full_messages).to include('CNPJ já está em uso')
+      end
+    end
+
+    context 'validate_CNPJ' do
+      it 'is valid' do
+        condo = build(:condo, registration_number: '38.352.640/0001-33')
+        condo.valid?
+        expect(condo.errors.full_messages).not_to include('CNPJ inválido')
+      end
+
+      it 'not valid' do
+        condo = build(:condo, registration_number: '88.99.555/4444-44')
+        condo.valid?
+        expect(condo.errors.full_messages).to include('CNPJ inválido')
+      end
+    end
+
+    context 'format' do
+      it 'valid' do
```

Comment:
De novo as descrições não explicam o que está sendo testado. Acho que esse context e it estão fora do escopo do validate cnpj.

Reply:
Corrigido!